lib/codegen-meta/src/gen_types.rs

```diff
@@ -0,0 +1,72 @@
+//! Generate sources with type info.
+//!
+//! This generates a `types.rs` file which is included in
+//! `lib/codegen/ir/types.rs`. The file provides constant definitions for the
+//! most commonly used types, including all of the scalar types.
+//!
+//! This ensures that Python and Rust use the same type numbering.
```

Comment:
This ensures that the metaprogram and the generated program see the same type numbering :-).